### ISSUE TITLE
Ubuntu PPA could use some TLC

### ISSUE THREAD

**Issue by RobH123** [sile-typesetter/sile#1851]
Looking at https://launchpad.net/~sile-typesetter/+archive/ubuntu/sile/+packages it says `This PPA currently publishes packages for Kinetic, Jammy, Impish, Hirsute, Groovy, Focal, and Bionic. `

**Lunar** was released in April this year. (Mantic is due in October.)

Groovy, Hirsute, Impish and Kinetic are no longer supported by Ubuntu according to https://wiki.ubuntu.com/Releases.

**Comment by alerque** [comment]
Yes, that note is a bit out of date. I actually dropped a couple of those when they went EOL before our last release, and started adding Lunar. Unfortunately I ran out of time before I hit the road for a month and not all the dependencies got built for Lunar. I'll deal with it when I can get back into my desktop where I have 40+ LXC virtual machines scripted for generating Ubuntu packages. That's a bit much for my travel laptop. Also my power went off at home so I can't get in remotely. I'm not looking forward to the refrigerator situation when I get home!

**Comment by RobH123** [comment]
Yes, I realise it's hard and complex work keeping up with all that. Thanks for even trying! I struggle just to keep my own tools on one system all up to date. (After being surprised at finding no Lunar PPA, I did manage to compile SILE from source and it ran so this isn't urgent for me now. Then what really messed me up was multiple versions of Lua installed when I went to install an external package -- that took me longer to sort out, and I suspect a PPA binary wouldn't have helped with that anyway. It was being told about `luarocks list` that helped there.)

**Comment by alerque** [comment]
Yes the situation with PPA packages using Lua is less than ideal because to date (to maintain compatibility with old Ubuntu LTS releases) I've been building then with the lowest common denominator Lua version. That isn't the default Lua on the newest release series, so that's confusing when you run LuaRocks to install a SILE extension module you have to match the Lua version SILE used (`sile --version` should show it at least). We should document this better, especially for Ubuntu. It isn't such a problem on other platforms where the default system Lua matches. In fact if you don't mind maybe pop open a new issue just for improving documentation on how to get Lua modules installed for the right Lua interpreter version. There are several things we could say about that...

**Comment by alerque** [comment]
The PPA is currently up to date and the instructions (inline errer messages, readme, manual) on how to deal with 3rd party modules and Lua version matching are much improved.

I welcome more specific issues if anything is still lacking.
